File: solutions/lua/triangle/1/triangle.lua
local triangle = {}

function triangle.kind(a, b, c)
  local err = 'Input Error'
  assert(a + b >= c, err)
  assert(b + c >= a, err)
  assert(a + c >= b, err)
  local seen = {}
  local count = 0
  for _, i in ipairs{a, b, c} do
    assert(i > 0, err)
    if not seen[i] then
      seen[i] = true
      count = count + 1
    end
  end
  if count == 1 then
    return 'equilateral'
  elseif count == 3 then
    return 'scalene'
  else
    return 'isosceles'
  end
end

return triangle

File: solutions/python/triangle/1/triangle.py
def valid(sides):
    return all(c > 0 for c in sides) and (sides[0] + sides[1] >= sides[2]) and (sides[1] + sides[2] >= sides[0]) and (sides[0] + sides[2] >= sides[1])

def equilateral(sides):
    return valid(sides) and len(set(sides)) == 1


def isosceles(sides):
    return valid(sides) and len(set(sides)) == 2 or len(set(sides)) == 1

1
def scalene(sides):
    return valid(sides) and len(set(sides)) == 3
